components/terrain/chunkmanager.cpp

```diff
@@ -164,8 +166,9 @@ std::vector<osg::ref_ptr<osg::StateSet> > ChunkManager::createPasses(float chunk
 
     float blendmapScale = mStorage->getBlendmapScale(chunkSize);
 
-    return ::Terrain::createPasses(useShaders, mSceneManager->getForcePerPixelLighting(),
-                                     mSceneManager->getClampLighting(), &mSceneManager->getShaderManager(), layers, blendmapTextures, blendmapScale, blendmapScale);
+    return ::Terrain::createPasses(useShaders,
+                                   mSceneManager->getForcePerPixelLighting(), mSceneManager->getClampLighting(), mSceneManager->getAccurateFog(),
+                                   &mSceneManager->getShaderManager(), layers, blendmapTextures, blendmapScale, blendmapScale);
```

Comment:
@zinnschlag @psi29a How do we feel about switching around where line breaks are when it isn't strictly necessary?

Reply:
Good catch! Can these be refactored into a plain struct? Similarly for `SkyManager` and `ShaderVisitor`.

Reply:
We're not the DirectX API. We don't put random structs that only get used in one place everywhere as it's the kind of abstraction which just makes things harder to read.

Reply:
Microsoft's VMS-like APIs are awful, yeah. But from my experience that mostly stems from *initializing* random structs, not passing them around. Also see that in `sky.hpp` there's already a random struct passed, then copied into that class' member variables.

Reply:
@AnyOldName3 what's worse, long getter chains or receiving a trivial settings struct by reference?

Reply:
To make it clear, I'm NOT telling you how to do things in your own project. Just that your way of doing things is pretty alien and I have trouble adjusting.